app/controllers/review_mapping_controller.rb

```diff
@@ -455,16 +482,36 @@ def start_self_review
   private
 
   def assign_reviewers_for_team(assignment_id, review_strategy, participants_hash)
-    if ReviewResponseMap.where(reviewed_object_id: assignment_id, calibrate_to: 0)
-                        .where('created_at > :time',
-                               time: @@time_create_last_review_mapping_record).size < review_strategy.reviews_needed
-
+    if check_eligibility(assignment_id,review_strategy)
       participants_with_insufficient_review_num = []
       participants_hash.each do |participant_id, review_num|
```

Comment:
Can you check whether [map](https://www.geeksforgeeks.org/ruby-array-map-function/) method can be used in this case?